From: Byeongsik Jeon <bsjeon@hanmail.net>
Subject: [PATCH v2 3/5] win32u: Remove the gamma value calibration code.
Message-Id: <20220421184922.394344-4-bsjeon@hanmail.net>
Date: Fri, 22 Apr 2022 03:49:20 +0900
In-Reply-To: <20220421184922.394344-1-bsjeon@hanmail.net>
References: <20220421184922.394344-1-bsjeon@hanmail.net>

Signed-off-by: Byeongsik Jeon <bsjeon@hanmail.net>
---
v2: no change.

This problem is caused by the FreeType hinting engine.
Here this code is unnecessary.

 dlls/win32u/font.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/dlls/win32u/font.c b/dlls/win32u/font.c
index 11111111111..11111111111 100644
--- a/dlls/win32u/font.c
+++ b/dlls/win32u/font.c
@@ -4887,7 +4887,7 @@ UINT init_font_options(void)
     char value_buffer[FIELD_OFFSET(KEY_VALUE_PARTIAL_INFORMATION, Data[20 * sizeof(WCHAR)])];
     KEY_VALUE_PARTIAL_INFORMATION *info = (void *)value_buffer;
     HKEY key;
-    DWORD i, val, gamma = 1400;
+    DWORD i, val, gamma = 1000;
     UINT dpi = 0;
 
     init_reg_keys();
@@ -4929,11 +4929,6 @@ UINT init_font_options(void)
         NtClose( key );
     }
 
-    /* Calibrating the difference between the registry value and the Wine gamma value.
-       This looks roughly similar to Windows Native with the same registry value.
-       MS GDI seems to be rasterizing the outline at a different rate than FreeType. */
-    gamma = 1000 * gamma / 1400;
-
     for (i = 0; i < 256; i++)
     {
         font_gamma_ramp.encode[i] = pow( i / 255., 1000. / gamma ) * 255. + .5;

-- 
2.40.0

